### PR TITLE
[HxSidebarBrand] LogoTemplate with context for collapsed state #264

### PR DESCRIPTION
**Pull request by Harvey1214**

### CODE REVIEW

**Review by hakenr — CHANGES_REQUESTED** [review]
I don't like the concept of `Logo` vs. `LogoShort`. We should stick with the context-supported render fragment where the `LogoTemplate` (please rename, this is the appropriate name adhering to our conventions, YES - BREAKING CHANGE) will be provided with context with `Collapsed` property (same as `HxSidebarItem.ContentTemplate`).

cc @crdo

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarBrandLogoTemplateContext.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Havit.Blazor.Components.Web.Bootstrap;
+
+/// <summary>
+/// Context provided to the <see cref="HxSidebarBrand.LogoTemplate" />.
+/// </summary>
+public class SidebarBrandLogoTemplateContext : SidebarCollapsedContext
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarBrandLogoTemplateContext.cs:12]: Not an inheritance.

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarCollapsedContext.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+/// <summary>
+/// Context indicating whether a <see cref="HxSidebar"/> is collapsed or expanded.
+/// </summary>
+public class SidebarCollapsedContext
+{
+	/// <summary>
+	/// Indicates whether a <see cref="HxSidebar"/> is collapsed or expanded.
+	/// </summary>
+	public bool Collapsed { get; set; }
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarCollapsedContext.cs:17]: We have to distinguish horizontal and vertical motion in the sidebar.
My proposal for property names:
* `SidebarCollapsed`
* `ItemExpanded`

The Sidebar is expanded by default, therefor `SidebarCollapsed` makes sense (`if (context.SidebarCollapsed) ...`).
The SidebarItem is collapsed by default therefor `ItemExpanded` makes sense...

What do you think?
cc @crdo @alexax578

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarTogglerTemplateContext.cs [hunk 3]
```diff
@@ -9,10 +9,7 @@ namespace Havit.Blazor.Components.Web.Bootstrap;
 /// <summary>
 /// Context provided to the <see cref="HxSidebar.TogglerTemplate" />.
 /// </summary>
-public class SidebarTogglerTemplateContext
+public class SidebarTogglerTemplateContext : SidebarCollapsedContext
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarTogglerTemplateContext.cs:12]: not an inheritance

**Review by hakenr — CHANGES_REQUESTED** [review]

File: Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarCollapsedContext.cs (hunk 2 above)
Comment (Harvey1214) [reply Havit.Blazor.Components.Web.Bootstrap/Navigation/SidebarCollapsedContext.cs:17]: Seems reasonable, will adjust the naming.

**Review by hakenr — APPROVED** [review]

**Pull request merged**